src/validator.spec.ts

```diff
@@ -1,82 +1,220 @@
+      const date = new Date(2018, 11, 30);
+
+      const result = validator({ value, customArgs: { date } });
+
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
+    });
+
+    it('should return succeeded validation when value is a valid Date object with year, month, days and hours earlier than customArgs date param', () => {
+      const value = new Date(2018, 11, 30, 10);
+      const date = new Date(2018, 11, 30, 15);
+
+      const result = validator({ value, customArgs: { date } });
+
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
+    });
+
+    it('should return succeeded validation when value is a valid Date object with year, month, days, hours and minutes earlier than customArgs date param', () => {
+      const value = new Date(2018, 11, 24, 10, 15);
+      const date = new Date(2018, 12, 24, 10, 33);
+
+      const result = validator({ value, customArgs: { date } });
+
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
+    });
+
+    it('should return succeeded validation when value is a valid Date object year, month, days, hours, minutes and seconds earlier than customArgs date param', () => {
+      const value = new Date(2018, 12, 30, 15, 33, 30);
+      const date = new Date(2018, 12, 30, 15, 33, 45);
+
+      const result = validator({ value, customArgs: { date } });
+
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
     });
-  });
 
-  it('should overwrite default message when it feeds value and calls to setErrorMessage', () => {
-    // Arrange
-    const value = 'test';
+    it('should return succeeded validation when value is a valid Date object year, month, days, hours, minutes, seconds and milliseconds earlier than customArgs date param', () => {
+      const value = new Date(2018, 12, 30, 15, 33, 45, 2);
+      const date = new Date(2018, 12, 30, 15, 33, 45, 6);
 
-    setErrorMessage('other message');
+      const result = validator({ value, customArgs: { date } });
 
-    // Act
-    const result = validator({ value });
+      expect(result).toEqual({
+        succeeded: true,
+        message: '',
+        type: 'PREVIOUS_DATE',
+      });
+    });
+
+    it('should return failed validation when value is a valid Date object later than customArgs date param', () => {
+      const value = new Date(2019, 11, 24, 10, 33, 30, 0);
+      const date = new Date(2018, 12, 30, 15, 33, 30, 0);
+
+      const result = validator({ value, customArgs: { date } });
+
+      expect(result).toEqual({
+        succeeded: false,
+        message: "Date isn't previous to the one provided.",
+        type: 'PREVIOUS_DATE',
+      });
+    });
+
+    it('should throw an error when it feeds value equals undefined', () => {
```

Comment:
It doesn't have to throw an error here. See previous comment

Reply:
Removed error Throw => cd08fccdc089f216eec3d2ac6576651f215cb698